SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -618,18 +618,16 @@ void USpatialNetDriver::OnMapLoaded(UWorld* LoadedWorld)
 
 	if (IsServer())
 	{
-		if (GlobalStateManager != nullptr)
+		if (GlobalStateManager != nullptr &&
+			!GlobalStateManager->IsReadyToCallBeginPlay() &&
+			StaticComponentView->HasAuthority(GlobalStateManager->GlobalStateManagerEntityId, SpatialConstants::STARTUP_ACTOR_MANAGER_COMPONENT_ID))
 		{
-			// Increment the session id, so users don't rejoin the old game.
+			// ServerTravel - Increment the session id, so users don't rejoin the old game.
 			GlobalStateManager->SetCanBeginPlay(true);
```

Comment:
If GSM isn't ready when the map is loaded, what's the expected behaviour here? When will this get re-called?

Reply:
GlobalStateManager::AuthorityChanged. When we gain authority over the STARTUP_ACTOR_MANAGER_COMPONENT_ID we call GlobalStateManager:: SetCanBeginPlay(true)

Reply:
What about the other three calls to the GSM?

Reply:
HandleStartupOpQueueing will call TriggerBeginPlay.
SetAcceptingPlayers is set to true in GlobalStateManager::AuthorityChanged when we gain authority over DEPLOYMENT_MAP_COMPONENT_ID
We don't want to call IncrementSessionID on first session